### ISSUE TITLE
Connecting to localhost MariaDB over SSH tunnel uses '::1' instead of localhost

### ISSUE THREAD

**Issue by naitsirch** [dbeaver/dbeaver#10410]
#### System information: 
- Debian Linux 10.6
- 7.2.5

#### Connection specification:
- 10.1.47-MariaDB-0+deb9u1 Debian 9.13 (running on a remote server)
- MariaDB
- SSH

#### Describe the problem you're observing:

I have setup a new connection over SSH. The SSH tunnel configuration test was successfull. On the remote system I want to connect to the database via 'localhost'.

But when I try to connect to the database, the following error occurs:
> Access denied for user 'web1268_d_rmp'@'::1' (using password: YES)
> Current charset is UTF-8. If password has been set using other charset, consider using option 'passwordCharacterEncoding'

The password only contains ASCII characters. So i can exclude an encoding issue.
But it seems like DBeaver is using the IPv6 loopback IP '::1' instead of my setting 'localhost'. Here's a screenshot from the configuration dialog and the error message:

![grafik](https://user-images.githubusercontent.com/1118790/99503686-cb65da00-297e-11eb-801c-141fe355668d.png)

The mysql user is only allowed for 'localhost' but not for '::1'. When I try to connect to the database on the remote client via native mariadb client I get the same error message when I use '::1':
> $ mysql -h '::1' -u web1268_d_rmp -p web1268_d_rmp
> Enter password: 
> ERROR 1045 (28000): Access denied for user 'web1268_d_rmp'@'::1' (using password: YES)

When I use 'localhost' it works fine:
> $ mysql -h 'localhost' -u web1268_d_rmp -p web1268_d_rmp
> Enter password: 
> Reading table information for completion of table and column names
> You can turn off this feature to get a quicker startup with -A
> 
> Welcome to the MariaDB monitor.  Commands end with ; or \g.
> ...

#### Steps to reproduce, if exist:

See above

**Comment by kseniiaguzeeva** [comment]
How I can see from the error message, ::1 is the user's host.
Please try to enter localhost in SSH local host field.
![Screenshot 2020-11-25 at 16 48 43](https://user-images.githubusercontent.com/51405061/100235993-229b1a00-2f3e-11eb-80d7-4c2b0b99a61a.png)

If this does not help, you need to add the ability to connect to MySQL with ::1

**Comment by naitsirch** [comment]
@kseniiaguzeeva I have tested your suggestion but it did not help. And I do not understand why you closed the issue. DBeaver uses the wrong hostname on the remote system.

Is it a bug or not?

**Comment by kseniiaguzeeva** [comment]
It is written user 'web1268_d_rmp'@'::1' in the error. So as I see, it is user's host, not database's. Could you check it?

If it is a bug from our side, of cause, the ticket will be reopened.

**Comment by naitsirch** [comment]
What do you mean by "it is user's host"?

And how can I check it?

**Comment by kseniiaguzeeva** [comment]
try to execute select host, user from mysql.user;
find your user name and the host for the user

Is it like in the error message?

**Comment by stollr** [comment]
@kseniiaguzeeva It is long ago, but today I have found the underlying reason and wanted to share the information ;-)

In MySQL and MariaDB the hostname "**localhost**" is used for UNIX socket connections. When the access happens through a ssh tunnel, the UNIX socket connection is not possible, thus it fallbacks to TCP connection (::1 or 127.0.0.1).

But in newer MySQL/MariaDB versions the access via TCP is disabled by default. This was true for our installation, too. When our admin enabled the user for external access, the connection was finally possible :-)

To check the accessability one has to look into the `user` table of the `mysql` database:

```
MariaDB [mysql]> SELECT User,Host FROM user WHERE User = 'stollr';
+-------------+-----------------+
| User        | Host            |
+-------------+-----------------+
| stollr      | %               | <- this is required for external access via TCP
| stollr      | localhost       |
+-------------+-----------------+
```

**Comment by mfeilen** [comment]
```
> MariaDB [mysql]> SELECT User,Host FROM user WHERE User = 'stollr';
> | stollr      | %               | <- this is required for external access via TCP
```
Bad suggestion - as it creates an open door to everyone when not properly firewalled. 
It seems the MariaDB driver is doing some kind of resolving at the wrong moment through the SSH tunnel, localhost is resolved as 127.0.0.1 / :::1 (IPv6) - which shouldn't. As user@127.0.0.1 is not the same as user@localhost connection is refused (which is normal behaviour).

Solution: add another user@127.0.0.1 / :::1 - then the tunnel is working.

**Comment by stollr** [comment]
Yeah, you are right. But if a SSH tunnel is required to connect to the DBMS it should be safe. Still, your comment is valid. Thanks for the addition.
